src/main/java/tdl/record/sourcecode/snapshot/file/SnapshotsFileWriter.java

```diff
@@ -45,14 +45,20 @@ public void takeSnapshot() {
             //try (ByteArrayOutputStream buff = createSnapshotAndStoreToByteArray()) {
             SnapshotFileSegment segment = new SnapshotFileSegment();
             segment.type = (snapshot instanceof KeySnapshot) ? SnapshotFileSegment.TYPE_KEY : SnapshotFileSegment.TYPE_PATCH;
-            segment.timestamp = TimeUnit.NANOSECONDS.toSeconds(timeSource.currentTimeNano());
+            segment.relativeTimestamp = TimeUnit.NANOSECONDS.toSeconds(timeSource.currentTimeNano());
+            segment.absoluteTimestamp = getTimestamp();
             segment.setData(snapshot.getData());
             IOUtils.write(segment.asBytes(), outputStream);
         } catch (IOException ex) {
             Logger.getLogger(SnapshotsFileWriter.class.getName()).log(Level.SEVERE, null, ex);
         }
     }
 
+    public int getTimestamp() {
+        Long unixTimestamp = System.currentTimeMillis() / 1000L;
```

Comment:
The time information should come from a timeSource. I always consider time to be an external dependency.
In this case we do not use this time information for scheduling or controlling threads so it is pure data and this data comes from outside.

I would create a class to represent the Metadata - information about recording and explicitly add it to the constructor:
```
       SourceCodeRecorder sourceCodeRecorder = new SourceCodeRecorder.Builder(..)
                .withMetadata(recordingTimestampMillis)
                .withTimeSource(new FakeTimeSource())
```

Reply:
When we add the header to the file we should also add a magic number for the file identifier.
I thing it should be SRCSTM

Reply:
sorry i misread the description